[nabonitasen] iP

nabonitasen opened the pull request:
# Barkley :dog:
> “One of the most important things I do daily that truly helps me make the most of my day is my daily “To-Do List” for both my business and personal life!” — Barbara Corcoran (from [Shark Tank](https://en.wikipedia.org/wiki/Shark_Tank))

Introducing Barkley :dog:, the _paw-fect_ personal assistant. He makes tracking your work
- Easy
- Quick
- Interactive

All you need to do is:
1. Download 
1. Double click the file
1. Add tasks!

Barkley is able to:
- [x] Track tasks
- [x] Track events
- [ ] Give reminders

Most importantly, he's _labrador-able_! What are you waiting for? We're _paw-sitive_ you'll **LOVE** working with Barkley :dog:!
##      
To learn more about how Barkley's code works, you can start by looking at the `main` method:
```ruby
public static void main(String[] args) {
    new Duke().run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

nabonitasen committed:
Rename, greet, exit

nabonitasen committed:
echo function

nabonitasen committed:
Add, List function

nabonitasen committed:
Mark/Unmark as done function

nabonitasen committed:
Todos, Events, Deadlines function

nabonitasen committed:
Automated Text UI Testing

nabonitasen committed:
Handle Errors function

nabonitasen committed:
Delete function

nabonitasen committed:
save function

nabonitasen committed:
date and time function

nabonitasen committed:
more OOP

nabonitasen committed:
add packages

nabonitasen committed:
build changes

nabonitasen committed:
Merge branch 'add-gradle-support'

nabonitasen committed:
update package + gradle

nabonitasen committed:
JUnit testing

nabonitasen committed:
Improve coding standard

nabonitasen committed:
Add JavaDoc

nabonitasen committed:
Add find function

nabonitasen committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/DataStorage.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/tasks/Task.java

nabonitasen committed:
Merge branch 'branch-level-9'

# Conflicts:
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/tasks/Task.java

nabonitasen committed:
Fix format

zhengyup commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,58 @@
+        while (!isExit) {
+            try {
+                String fullCommand = ui.readCommand();
+                ui.showLine(); // show the divider line ("_______")
```
I like this minimal comment !

zhengyup commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,58 @@
+            try {
+                String fullCommand = ui.readCommand();
+                ui.showLine(); // show the divider line ("_______")
+                Command c = Parser.parse(fullCommand, tasks);
```
could this name be more self explanatory?

zhengyup commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,150 @@
+
+        } else if (fullCommand.contains("event")) {
+            String taskInfo = fullCommand.substring(5);
+            if (taskInfo.isBlank() || !fullCommand.contains("/from")
+                    || !fullCommand.contains("/to") || !fullCommand.contains("event ")
+                    || fullCommand.substring(fullCommand.indexOf("/from") + 5).isBlank()
+                    || fullCommand.substring(fullCommand.indexOf("/to") + 3).isBlank()) {
```
could there be a way to shorten this expression ? maybe you can attempt to try creating the event and catching any exceptions thrown?

zhengyup commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,16 @@
+/**
+ * Represents exceptions unique to the application.
+ */
+public class DukeException extends Exception {
```
Could this class be abstracted more into separate kinds of exceptions to be thrown ?

zhengyup reviewed (commented):
LGTM! Few nits to fix. Good job!

nabonitasen committed:
Add checkstyle

nabonitasen committed:
Add GUI

nabonitasen committed:
Add assertions

Ensure inputs are not empty with assertions.

nabonitasen committed:
Improve Code Quality + FIx bugs

Updates to list do not save until user sends 'bye' and number of tasks shown not accurate after 'delete'.

Storage should constantly update in case the user closes the application without sending 'bye' and number of tasks shown should be accurate.

Lets:
# Update storage after every command.
# Show accurate number of tasks after 'delete'.

nabonitasen committed:
Merge pull request #1 from nabonitasen/branch-A-Assertions

Add assertions

nabonitasen committed:
Merge branch 'master' into branch-A-CodeQuality

nabonitasen committed:
Merge pull request #2 from nabonitasen/branch-A-CodeQuality

Improve Code Quality + Fix bugs

nabonitasen committed:
Set up CI

nabonitasen committed:
Delete I/O redirection tests

nabonitasen committed:
Add update function + Update JavaDoc

Users cannot update tasks without deleting and re-adding.

Let's add an update function.

nabonitasen committed:
Fix bugs

Only first word of description input is recorded. Let's fix this to record whole input.

nabonitasen committed:
Add Ui.png and edit UI formatting

nabonitasen committed:
Add clear function

nabonitasen committed:
Add user guide

nabonitasen committed:
Update user guide further

nabonitasen committed:
Edit build.gradle to allow JAR file to run on multiple OS'
